File: 888/solution.py
# two sum variant

class Solution:
    def fairCandySwap(self, A: List[int], B: List[int]) -> List[int]:
        sum1 = sum(A)
        sum2 = sum(B)
        diff = (sum1 - sum2) // 2
        s = set(A)
        
        for x in B:
            target = x + diff
            if target in s:
                return [target, x]
            
        return []

File: 888/solution.java
// two sum variant

class Solution {
    public int[] fairCandySwap(int[] A, int[] B) {
        int sum1 = Arrays.stream(A).sum();
        int sum2 = Arrays.stream(B).sum();
        int diff = (sum1 - sum2) / 2;
        Set<Integer> s = new HashSet();
        for(int elem : A) {
            s.add(elem);
        }
        
        for(int x : B) {
            int target = x + diff;
            if(s.contains(target)) {
                return new int[] {target, x};
            }
        }
        
        return new int[] {};
    }
}
